app/api/authentication.py

```diff
@@ -70,3 +73,45 @@ def auth_status():
             'user_id': getattr(current_user, 'user_id', None),
         },
     }), constants.api.SUCCESS_CODE
+
+
+def oidc_request_loader():
+    """
+    flask_login request loader for OpenID Connect.
+    """
+    if current_user.is_authenticated:
+        return
+
+    from modern_paste import oidc
+    info = None
+    if flask.g.oidc_id_token is not None:
+        # Auth succeeded with flask-oidc OIDC client flow
+        username = oidc.user_getfield('sub')
+        try:
+            login(database.user.get_user_by_username(username))
+            return
+        except UserDoesNotExistException:
+            info = oidc.user_getinfo(['sub', 'name', 'email'])
+
+    elif flask.request.headers.get('Authorization', '').startswith('Bearer '):
+        # An OAuth2 Bearer Token was sent
+        token = flask.request.headers['Authorization'].split(' ', 1)[1].strip()
+        valid = oidc.validate_token(token, config.AUTH_OIDC_SCOPE)
+        if valid is True:
+            username = flask.g.oidc_token_info['sub']
+            try:
+                login(database.user.get_user_by_username(username))
+            except UserDoesNotExistException:
+                info = oidc.user_getinfo(['sub', 'name', 'email'], token)
+        else:
+            return flask.jsonify(constants.api.AUTH_FAILURE), constants.api.AUTH_FAILURE_CODE
```

Comment:
This function isn't an API handler, what is the meaning of returning an API response here?